games-action/dave_gnukem/patches/dave_gnukem-1.0.patchset

```diff
@@ -0,0 +1,149 @@
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Sat, 28 Dec 2019 13:42:47 +0000
+Subject: Allows launch from deskbar
+
+
+diff --git a/src/main.cpp b/src/main.cpp
+index 183b95a..bd3600e 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -38,6 +38,11 @@ Copyright (C) 1995-2018 David Joffe
+ #include "sys_log.h"
+ #include "sys_error.h"
+ 
++#ifdef __HAIKU__
++#include <unistd.h>
++#include <libgen.h>
++#endif
++
+ #ifdef __APPLE__
+ #include <mach-o/dyld.h>//_NSGetExecutablePath
+ #include <sys/stat.h>//For djFolderExists stuff
+@@ -114,6 +119,11 @@ CMenu mainMenu ( "main.cpp:mainMenu" );
+ // This is the 'main' function. The big cheese.
+ int main ( int argc, char** argv )
+ {
++	
++#ifdef __HAIKU__
++	chdir(dirname(argv[0]));
++#endif
++
+ 	// Check commandline args
+ 	bool bfullscreen = false;
+ 	bool b640 = false;
+-- 
+2.19.0
+
+
+From 3f4c0d461a3ab51a3bc95f147ba4ce5bc80a1487 Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Sat, 28 Dec 2019 13:45:45 +0000
+Subject: Fixed settings directory
+
+
+diff --git a/src/djstring.cpp b/src/djstring.cpp
+index 76237f9..2ba7187 100644
+--- a/src/djstring.cpp
++++ b/src/djstring.cpp
+@@ -22,6 +22,11 @@ Copyright (C) 1998-2018 David Joffe
+ #include <errno.h>//mkpath
+ #endif
+ 
++#ifdef __HAIKU__
++#include <Path.h>
++#include <FindDirectory.h>
++#endif
++
+ char *djStrDeepCopy( const char * src )
+ {
+ 	if (!src)
+@@ -284,6 +289,12 @@ std::string djGetFolderUserSettings()
+ #ifdef WIN32
+ 	std::string s = getenv("APPDATA");
+ 	djAppendPathS(s, "DaveGnukem");
++#elif __HAIKU__
++	std::string s;
++	BPath * s_path = new BPath();
++	find_directory(B_USER_SETTINGS_DIRECTORY, s_path);
++	djAppendPathS(s, s_path->Path());
++	djAppendPathS(s, ".gnukem");
+ #else
+ 	std::string s = getenv("HOME");
+ 	djAppendPathS(s, ".gnukem");//?
+-- 
+2.19.0
+
+
+From 2b1e1b003d87a61e9fb5b9b1bfbf999c51847f8f Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Sat, 28 Dec 2019 13:49:27 +0000
+Subject: Added required lib for FindDirectory.h and Path.h
+
+
+diff --git a/Makefile b/Makefile
+index 8fc1184..3ffde58 100644
+--- a/Makefile
++++ b/Makefile
+@@ -25,7 +25,7 @@ CCFLAGS = -Wall -Wno-switch -DDEBUG $(INCLUDEDIRS)
+ #Release version:
+ #CCFLAGS = -O -Wall -I/usr/local/include -DHAVE_SOUND $(INCLUDEDIRS)
+ 
+-LIBS = -lSDL -lSDLmain -lSDL_mixer -lpthread 
++LIBS = -lSDL -lSDLmain -lSDL_mixer -lpthread -lbe
+ BIN = davegnukem
+ 
+ 
+-- 
+2.19.0
+
+
+From 0235ee1bcfbf50553f86a3b56091ebd9e8289c1a Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Sun, 29 Dec 2019 10:42:00 +0000
+Subject: changed deskbar launch fix to work now
+
+
+diff --git a/src/main.cpp b/src/main.cpp
+index bd3600e..8321613 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -121,7 +121,11 @@ int main ( int argc, char** argv )
+ {
+ 
+ #ifdef __HAIKU__
+-	chdir(dirname(argv[0]));
```

Comment:
At a quick look it looks OK (haven't checked yet), but could you combine the patch for this into 1 patch to launch from Tracker and Deskbar?

Reply:
Done!